src/layout/output.cpp

```diff
@@ -19,12 +19,18 @@
 namespace chopper::layout
 {
 
-void write_user_bins_to(std::vector<std::string> const & filenames, std::ostream & stream)
+void write_user_bins_to(std::vector<std::vector<std::string>> const & filenames, std::ostream & stream)
 {
     stream << chopper::prefix::meta_chopper_user_bins_start << '\n';
     size_t counter{};
-    for (auto const & filename : filenames)
-        stream << seqan::hibf::prefix::meta_header << counter++ << ' ' << filename << '\n';
+    for (auto const & filenames_of_user_bin : filenames)
+    {
+        stream << seqan::hibf::prefix::meta_header << counter++ << ' ';
+        for (std::string const & filename : filenames_of_user_bin)
+            stream << filename << ' ';
+        stream.seekp(static_cast<int>(stream.tellp()) - 1); // overwrite last ' '
```

Comment:
```suggestion
        stream.seekp(stream.tellp() - 1); // overwrite last ' '
```

`tellp` should return `typename std::ostream::pos_type`, and `seekp` takes a `typename std::ostream::pos_type`.

But `seekp` might be expensive, depending on what the kernel makes of that, no?

Reply:
`stream.tellp()) - 1` didn't compile because you cannot substract from `typename std::ostream::pos_type`.
But given your 2nd comment we don't need this at all :)